integration-tests/tests/integration_test.rs

```diff
@@ -12,13 +12,15 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use crate::test_utils::{
+mod test_utils;
```

Comment:
JFYI I am likely to use these `test_utils` to implement an mdbook doctest macro per #797.

Reply:
> JFYI I am likely to use these `test_utils` to implement an mdbook doctest macro per #797.

Thanks for the heads up. I turned it back into draft because I'm seeing flaky integration tests on some of the architectures. I don't know the cause. A possible cause could be a compiler bug (and in fact the error messages literally always say that it's a compiler bug). However, before giving up on having reliable non-compiler-crashing integration tests, I thought I'd like to see if I could get the tests to run just via `cargo test` without a `cargo build` needed first. That means turning the tests that need binaries to work into Rust "integration tests" and that's the work you can see being partially done here.

I wish I could do it all locally, but I haven't got a way of running github actions across multiple architectures locally.

I *can* reproduce the flaky test failures on beta and nightly Rust channels locally. Anyway, that's where it's at.

Obviously if you merge #797 first I will have to deal. Otherwise we'll find some way of sharing the code that needs to be shared between integration tests and the mdbook, I'm sure.